src/convertToObject.js

```diff
@@ -6,7 +6,26 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const arraySorceString = sourceString
+    .split(';')
+    .map((string) => string.trim().split(':'));
+
+  const filteredSorceString = arraySorceString
+    .filter((innerArr) => innerArr.length === 2)
+    .map((element) => element.map((item) => item.trim()));
+
+  // console.log(filteredSorceString);
+
+  const convertingObject = filteredSorceString.reduce((acc, arrayElement) => {
```

Comment:
Instead of using 'reduce' to construct an object, consider using 'Object.fromEntries' which is a more declarative and modern approach to convert an array of key-value pairs into an object.